reframe/core/pipeline.py

```diff
@@ -1459,6 +1459,11 @@ class RunOnlyRegressionTest(RegressionTest):
     module.
     '''
 
+    def _rfm_init(self, *args, **kwargs):
+        super()._rfm_init(*args, **kwargs)
+        if not os.path.exists(os.path.join(self._prefix, 'src')):
+            self.sourcesdir = None
+
```

Comment:
As pointed out by @teojgo, it is wrong to set it here. Also why don't we generalise this behaviour and implement it once in the `RegressionTest()`? if `src/` exists, it becomes the default otherwise `sourcesdir` is set to `None`. This can be done in the `_rfm_init()` of `RegressionTest`. We will need to update the documentation likewise and perhaps a new unit test or adapt an existing one.

Reply:
Oh, I should remove this.